src/Aspire.Hosting/JavaScript/JavaScriptExtensions.cs

```diff
@@ -0,0 +1,88 @@
+    {   
+        scriptPath ??= "node";
+        args ??= [];
+        string[] effectiveArgs = [scriptPath, .. args];
+        workingDirectory ??= Path.GetDirectoryName(scriptPath)!;
+        workingDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, workingDirectory));
+
+        var resource = new JavaScriptAppResource(name, scriptPath, workingDirectory, effectiveArgs);
+
+        return builder.AddResource(resource)
+                      .WithJavaScriptAppDefaults();
+    }
+
+    /// <summary>
+    /// Adds a node application to the application model. Executes the npm command with the specified script name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="scriptName">The npm script to execute. Defaults to "start".</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<JavaScriptAppResource> AddJavaScriptCLIApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string? scriptName = null, string[]? args = null)
+    {   
+        args ??= [];
+        scriptName ??= "npm";
+        string[] allArgs;
+
+        // Check if scriptName is "npm" and adjust allArgs accordingly
+        if (string.Equals(scriptName, "npm", StringComparison.OrdinalIgnoreCase))
+        {
+            allArgs = args is { Length: > 0 }
+                ? new string[] { "run", scriptName, "--" }.Concat(args).ToArray()
+                : new string[] { "run", scriptName };
+        }
+        else
+        {
+            // If scriptName is not npm, use developer config or an empty array
+            allArgs = args ?? Array.Empty<string>();
+        }
+
+        workingDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, workingDirectory));
+
+        var resource = new JavaScriptAppResource(name, scriptName, workingDirectory, allArgs);
+
+        return builder.AddResource(resource)
+                      .WithJavaScriptAppDefaults();
+    }
+
+    // this needs tested only for node since
+    // other runtimes may use a different pattern for env configuration
+     private static IResourceBuilder<JavaScriptAppResource> WithJavaScriptAppDefaults(this IResourceBuilder<JavaScriptAppResource> builder)
+    {
+        var environment = builder.ApplicationBuilder.Environment;
+        Console.WriteLine("Resource Command: " + builder.Resource.Command);
```

Comment:
I'm sure this is just temporary debug output but we don't do `Console.WriteLine(...)` in this methods. They generally don't have logging at all. These commands run before our DI container is spun up so we can't provide them a `ILogger<T>` either.